Azure Cloud Control Manager - nil pointer dereference and Crashbackloop

**What happened**

When we start the Azure Cloud Control Manager Pod, it seems to run fine at first, but a little ways through the startup, we are seeing this nil pointer error in the log and then the pod starts a crashbackloop. It is actually able to perform its jobs eventually (provisioning load balancers, dynamic provisioning, etc.) in between failures, but fails at the same spot each time and restarts in a loop.

```
`E0306 16:09:43.558101       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 120 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x1b1b2e0, 0x2fc89a0)
        /go/pkg/mod/k8s.io/apimachinery@v0.21.0-alpha.0.0.20201209085528-15c5dba13c59/pkg/util/runtime/runtime.go:74 +0x92
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
        /go/pkg/mod/k8s.io/apimachinery@v0.21.0-alpha.0.0.20201209085528-15c5dba13c59/pkg/util/runtime/runtime.go:48 +0x82
panic(0x1b1b2e0, 0x2fc89a0)
        /usr/local/go/src/runtime/panic.go:679 +0x1b2
sigs.k8s.io/cloud-provider-azure/pkg/provider.(*delayedRouteUpdater).updateRoutes(0xc0006846f0)
        /go/src/sigs.k8s.io/cloud-provider-azure/pkg/provider/azure_routes.go:153 +0x1d0
sigs.k8s.io/cloud-provider-azure/pkg/provider.(*delayedRouteUpdater).run.func1(0xc000123530, 0xc00017c6c0, 0xc000566688)
        /go/src/sigs.k8s.io/cloud-provider-azure/pkg/provider/azure_routes.go:97 +0x2a
k8s.io/apimachinery/pkg/util/wait.runConditionWithCrashProtection(0xc0001237b8, 0xc000123500, 0x0, 0x0)
        /go/pkg/mod/k8s.io/apimachinery@v0.21.0-alpha.0.0.20201209085528-15c5dba13c59/pkg/util/wait/wait.go:211 +0x6f
k8s.io/apimachinery/pkg/util/wait.WaitFor(0xc000050dc0, 0xc000ba5fb8, 0xc0000aa3c0, 0x0, 0x0)
        /go/pkg/mod/k8s.io/apimachinery@v0.21.0-alpha.0.0.20201209085528-15c5dba13c59/pkg/util/wait/wait.go:541 +0x159
k8s.io/apimachinery/pkg/util/wait.PollUntil(0x6fc23ac00, 0xc0001237b8, 0xc0000aa240, 0x0, 0x0)
        /go/pkg/mod/k8s.io/apimachinery@v0.21.0-alpha.0.0.20201209085528-15c5dba13c59/pkg/util/wait/wait.go:492 +0xd6
k8s.io/apimachinery/pkg/util/wait.PollInfinite(0x6fc23ac00, 0xc0001237b8, 0x0, 0x0)
        /go/pkg/mod/k8s.io/apimachinery@v0.21.0-alpha.0.0.20201209085528-15c5dba13c59/pkg/util/wait/wait.go:464 +0x9f
k8s.io/apimachinery/pkg/util/wait.PollImmediateInfinite(0x6fc23ac00, 0xc0001237b8, 0x0, 0x0)
        /go/pkg/mod/k8s.io/apimachinery@v0.21.0-alpha.0.0.20201209085528-15c5dba13c59/pkg/util/wait/wait.go:481 +0x75
sigs.k8s.io/cloud-provider-azure/pkg/provider.(*delayedRouteUpdater).run(0xc0006846f0)
        /go/src/sigs.k8s.io/cloud-provider-azure/pkg/provider/azure_routes.go:96 +0x52
created by sigs.k8s.io/cloud-provider-azure/pkg/provider.(*Cloud).InitializeCloudFromConfig
        /go/src/sigs.k8s.io/cloud-provider-azure/pkg/provider/azure.go:567 +0xa72

I0306 16:09:43.558228       1 azure_routes.go:406] CreateRoute: route created. clusterName="k8s" instance="gfmdev-db1-1-vm" cidr="10.244.2.0/24"
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1877230]goroutine 120 [running]:`
```

**What you expected to happen**:

Either no error or fail gracefully with a meaningful error message.

**How to reproduce it**:

Deploy the Cloud Control Manager as a pod. Our Azure.json file is configured as the below:

```
{
    "cloud": "AzureUSGovernmentCloud",
    "tenantId": "XYZ",
    "subscriptionId": "XYZ",
    "aadClientId": "XYZ",
    "aadClientSecret": "XYZ",
    "resourceGroup": "DEVELOPMENT",
    "location": "usgovvirginia",
    "vmType": "standard",
    "subnetName": "dev-private-subnet-0",
    "securityGroupName": "dev-private-nsg",
    "vnetName": "dev-vnetwork",
    "vnetResourceGroup": "DEVELOPMENT",
    "cloudProviderBackoff": false,
    "useInstanceMetadata": true,
    "loadBalancerSku": "standard",
    "maximumLoadBalancerRuleCount": 250,
}

```
**Anything else we need to know?**

We are running K8s (not AKS) on Azure Gov VMs. Everything seems to be functioning properly and we are even able to use CCM to auto provision LBs and for dynamic volume provisioning, but the continual crashbackloop makes it take a lot longer to provision and is not sustainable longterm. We ran into this same issue when running on Azure Public Cloud as well.

**Environment**:
- Kubernetes version (use `kubectl version`): 1.20.4 
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others: 
            - GoVersion - go1.15.8
            - Azure CCM Version - v0.7.2

--- a/pkg/provider/azure_routes.go
+++ b/pkg/provider/azure_routes.go
@@ -150,7 +150,7 @@ func (d *delayedRouteUpdater) updateRoutes() {
 	// reconcile routes.
 	dirty := false
 	routes := []network.Route{}
-	if routeTable.Routes != nil {
+	if routeTable.RouteTablePropertiesFormat != nil && routeTable.RouteTablePropertiesFormat.Routes != nil {
 		routes = *routeTable.Routes
 	}
 	onlyUpdateTags := true

